### ISSUE TITLE
about your losss function

### ISSUE THREAD

**Issue by yangdongchao** [Pilhyeon/BaSNet-pytorch#19]
you use following code to get predict probability 
score_base = self.softmax(score_base)
score_supp = self.softmax(score_supp)

but your loss function is choose BCE loss.

I just cannot understand, this is a multiple classification problems, why you use softmax to get probability? mabey you should use sigmoid function, because every class may be occur.

**Comment by yangjiangeyjg** [comment]
> 
> 
> you use following code to get predict probability
> score_base = self.softmax(score_base)
> score_supp = self.softmax(score_supp)
> 
> but your loss function is choose BCE loss.
> 
> I just cannot understand, this is a multiple classification problems, why you use softmax to get probability? mabey you should use sigmoid function, because every class may be occur.

I also cannot understand this problem. how did you solve it.

**Comment by yangdongchao** [comment]
In fact, when it calculate loss, it use the normalized true label (e.g. [1,1,0] ---> [0.5,0.5,0]), so it is also correct. But the drawback is you must set a thres_hold according to your experiments, such as thres_hold=0.2, indicates the class is occur.


| |
15087581161
|
|
***@***.***
|
签名由网易邮箱大师定制
On 9/22/2021 ***@***.***> wrote：

you use following code to get predict probability
score_base = self.softmax(score_base)
score_supp = self.softmax(score_supp)

but your loss function is choose BCE loss.

I just cannot understand, this is a multiple classification problems, why you use softmax to get probability? mabey you should use sigmoid function, because every class may be occur.

I also cannot understand this problem. how did you solve it.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by yangjiangeyjg** [comment]
> 
> In fact, when it calculate loss, it use the normalized true label (e.g. [1,1,0] ---> [0.5,0.5,0]), so it is also correct. But the drawback is you must set a thres_hold according to your experiments, such as thres_hold=0.2, indicates the class is occur. | | 15087581161 | | ***@***.*** | 签名由网易邮箱大师定制 On 9/22/2021 ***@***.***> wrote： you use following code to get predict probability score_base = self.softmax(score_base) score_supp = self.softmax(score_supp) but your loss function is choose BCE loss. I just cannot understand, this is a multiple classification problems, why you use softmax to get probability? mabey you should use sigmoid function, because every class may be occur. I also cannot understand this problem. how did you solve it. — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

Thanks!
